department-of-veterans-affairs/vets-website
# Check-in translation PoC

## Description

The check-in experience group has been asked to translate the check-in and pre-check-in apps by May 2. We have a working PoC for the whole app, but this PR provides the functionality to just the error page. The purpose is to provide a more streamlined PR for the frontend review group to review and request the required dependencies:

- [i18next](https://www.i18next.com)
- [react-i18next](https://react.i18next.com)
- [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector)
- [i18next-resources-to-backend](https://github.com/i18next/i18next-resources-to-backend)

react-i18next was chosen because of its adoption in the React community, and due to its [recommendation in past platform research](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content-localization/development/Decision-Log-I18N-JS-Libraries.md)

To test locally:

- Start the mock API: `yarn mock-api --responses src/applications/check-in/api/local-mock-api/index.js`
- Run the app: `yarn watch --env entry=check-in`
- Go to the error page: http://localhost:3001/health-care/appointment-check-in/error

You should see the app in english with English / Español buttons at the bottom. If you visit with your browser language set to Spanish, it should automatically select Spanish as the language.

NB: the translations in the file are machine translated for the PoC, but translation capability is behind a feature flag and will not be released to the public until official translations have been obtained & reviewed.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#38309


## Testing done

manual testing, updated unit and e2e tests

## Screenshots

![Screen Shot 2022-03-21 at 4 08 58 PM](https://user-images.githubusercontent.com/101649/159371640-56243dcb-0939-46b3-bc08-2e15527a4c95.png)
